src/core/audio/AudioRecorder.h

```diff
@@ -33,11 +34,11 @@ struct AudioRecorder final {
 
     bool start(fs::path const& file);
     void stop();
-    bool isRecording() const;
+    [[nodiscard]] bool isRecording() const;
     std::vector<DeviceInfo> getInputDevices() const;
 
 private:
-    Settings& settings;
+    [[ maybe_unused ]] Settings& settings;
```

Comment:
I'm not sure marking those unused member refs/pointer makes a lot of sense. Maybe make two commits: the first with the other changes, and the second one actually removing the unused member refs/pointers.

Reply:
Will do! I wasn't sure if I should have put `[[ maybe_unused ]]`, or instead, removed the private member, and removed it from constructor's member initializer list, and also removed its parameter from the constructor. I'll make a second commit with the removed unused member refs / pointers.